Source/Core/DolphinQt2/WiiUpdate.cpp

```diff
@@ -74,46 +119,25 @@ void PerformOnlineUpdate(const std::string& region, QWidget* parent)
               true);
           return !was_cancelled.IsSet();
         },
-        region);
+        std::forward<Args>(args)...);
     Core::QueueHostJob([&dialog] { dialog.close(); }, true);
     return res;
   });
 
   dialog.exec();
+  Core::HostDispatchJobs();
```

Comment:
Yes, they're added to a std::queue and processed in order when calling HostDispatchJobs. It's indeed strange that the queue isn't empty before returning too, because dialog.exec() only returns after the dialog close job is processed, and that job is the last one in the queue... I'll try it again, maybe it's not needed anymore.